File: solutions/javascript/src/2022/utils/load.js
import exp from 'constants';
import * as fs from 'fs'
import path from 'path';

const resources2022BasePath = "../../../../resources/2022";

export function getInput(dayStr, sampleNumber) {
    const filename = `input.${sampleNumber}.txt`;
    const relativePath = `${resources2022BasePath}/${dayStr}/${filename}`;

    const fileStringContent = readFileToString(relativePath);

    return parseFileContent(fileStringContent);
}

export function getExampleInput(dayStr, sampleNumber) {
    const filename = `example.${sampleNumber}.txt`;
    const relativePath = `${resources2022BasePath}/${dayStr}/${filename}`;

    const fileStringContent = readFileToString(relativePath);

    return parseFileContent(fileStringContent);
}

export function parseFileContent(fileContent) {
    const rows = fileContent.split("\r\n");

    return rows;
}

export function readFileToString(relativeFilePath) {
    let absolutePath = path.resolve(relativeFilePath);
    return fs.readFileSync(absolutePath).toString();
}

File: solutions/python/src/2022/utils/load.py
import os

INPUT_BASE = "../../../../../resources/2022"

def load_example_input(day: str, inputNumber: int) -> str:
    curr = os.path.dirname(__file__)
    filename = f"example.{inputNumber}.txt"
    filepath = os.path.join(curr, INPUT_BASE, day, filename)
    
    return load_file(filepath)
    
def load_input(day: str, inputNumber: int) -> str:
    curr = os.path.dirname(__file__)
    filename = f"input.{inputNumber}.txt"
    filepath = os.path.join(curr, INPUT_BASE, day, filename)
    
    return load_file(filepath)
    
def load_file(filepath: str) -> str:
    
    with open(filepath, 'r') as f:
        data = f.read()
        
    return data

def as_rows(file_content: str) -> list[str]:
    
    return file_content.split("\n")
